doc/source/tempfile.rst

```diff
@@ -0,0 +1,82 @@
+Temporary Files
+===============
+
+Ray will produce some temporary files during running.
+They are useful for logging, debugging & sharing object store with other programs.
+
+Location of Temporary Files
+---------------------------
+
+First we introduce the concept of a session of Ray.
+
+A session contains a set of processes. A session is created by executing
+``ray start`` command or call `ray.init()` in a Python script and ended by
+executiong ``ray stop`` or call `ray.shutdown()`.
+
+For each session, Ray will create a *root temporary directory* to place all its
+temporary files. The path is `/tmp/ray_{ray_pid}@{datetime}` by default.
```

Comment:
I am not sure if I would like to see `@` in the directory name. Also, should we maybe have `ray_{datetime}_{pid}`? Then sorting by name would sort also by time?

Reply:
I use `@` because there are too many `_`s & `-`s in the datetime string and it looks not nice for me. `@` also implies 'at' a certain timestamp and is supported in most platforms without escape charactors like `\`.

Reply:
`ray_pid` is the pid of which process? In one session, there will be many processes, right? 
Also, not a big deal, but I'd suggest put these temp dirs under one big root, like `/tmp/ray/${session_root}`. This will make these dirs easier to manage.